### DISCUSSION TITLE
Will the next version of lxqt be 0.17 or 1.0?

### DISCUSSION THREAD

**Discussion by telmobss1** [lxqt/lxqt#1945] [Q&A] [answered]
The versioning of lxqt looks like Elementaryos

What does lxqt need to be considered fully usable, both in terms of features and stability?

**Comment by tsujan** [comment]
LXQt *is* fully usable.

Yes, we should change the version to 1.0 at some point. I think the best time is when Qt6 is ready for use and we port LXQt to Qt6.

At least under Linux, 0.X doesn't mean "unstable". There are so many rock solid apps with versions like 0.X.Y.

**Reply by stefonarch** [reply]
I agree, 1.0.1 looks _slightly_ better than 0.16.1

**Reply by maverick74** [reply]
> I think the best time is when Qt6 is ready for use and we port LXQt to Qt6.

As just a user, this makes perfect sense to me :)

**Comment by stefonarch** [comment, marked as answer]
Just to answer: next version released (in april 2021) is 0.17

**Reply by tsujan** [reply]
> Just to answer: next version released (in april 2021) is 0.17

And will be released in a week.

**Reply by tsujan** [reply]
We should wait to see if Qt6 will be ready for real use and without Qt5 compatibility packages before November. Or we might agree on 1.0.0 even if it isn't.

**Reply by luis-pereira** [reply]
LXQt is ready for a 1.0 release, IMO. It's just a number anyway. 
But.... not this one. We are too close to the release date.

**Reply by tsujan** [reply]
> IMO. It's just a number anyway.

I agree completely.

**Reply by stefonarch** [reply]
> We should wait to see if Qt6 will be ready for real use and without Qt5 compatibility packages before November. Or we might agree on 1.0.0 even if it isn't.

Any news on that? Release date Novemeber?

**Reply by tsujan** [reply]
> Release date Novemeber?

Yes. like last year.

**Reply by tsujan** [reply]
As for the version, I'll go with any decision that might be taken until November. Since Qt6 still isn't usable to us (neither to KDE), I suggest 0.18 but 1.0.0 is equally acceptable.

**Comment by tsujan** [comment]
Dear LXQt members,

Any objection to bumping the version to 1.0.0 for the next release in November?

I'm not sure whether the April release (2022) could be based on Qt6 either. Provided that Qt ≥ 6.2.0 will be good, we'll also need to wait for KF6.

**Reply by luis-pereira** [reply]
No objections to 1.0.0.
If KF6 is available by that time, we should go for it.

**Reply by tsujan** [reply]
> If KF6 is available by that time, we should go for it.

I agree. Qt6 itself seems to be ready (→ https://github.com/lxqt/lxqt/discussions/1901#discussioncomment-1413077). Supporting Qt 5.15 will also be possible (see https://doc.qt.io/qt-6/cmake-qt5-and-qt6-compatibility.html#mixing-qt-5-and-qt-6) but I'm not sure it'll be worth cluttering the code.

**Reply by stefonarch** [reply]
As I manage atm the twitter and mastodon (testing) accounts: tell me when 1.0.0 is confirmed :)

**Reply by tsujan** [reply]
> tell me when 1.0.0 is confirmed 

It will be 1.0.0 — there was no objection. I'll bump versions while updating changelogs, like with previous releases.

**Reply by maverick74** [reply]
I'm sorry enter in a talk to which I'm not "invited to"... I just wanted to clarify: we'll finally jump to 1.0 which will coincide with the jump to the use of Qt6 (and if if the stars align  also with KF6)? Is that it?

**Reply by tsujan** [reply]
> we'll finally jump to 1.0 which will coincide with the jump to the use of Qt6...?

No! We'll bump the version to 1.0.0 in November and make a release against Qt 5.15. As mentioned elsewhere, we can't use Qt6 yet — mainly because of KF6 but we should also make sure that Qt6 is the default Qt version in rolling distros (not just Arch).

**Reply by maverick74** [reply]
Oh. I see. Thanks for clarifying that.
For a stable release 5.15 is always a more solid base anyway

**Reply by tsujan** [reply]
> For a stable release 5.15 is always a more solid base anyway

Good point.

**Reply by tsujan** [reply]
I'll start to make pre-release PRs (changelog updates and version bumps) soon and will merge them in the first week of November. Before that, 2 or 3 PRs should be reviewed and merged (I'll make one for lxqt-themes). Translation PRs can come and be merged, as before.

I'll add a comment here before starting to merge pre-release PRs (none is made yet).

**Reply by tsujan** [reply]
Something like this will be the announcement, which I hope I could add to https://github.com/lxqt/lxqt/releases tomorrow:


The LXQt team is proud to announce the release of LXQt 1.0.0, the Lightweight Qt Desktop Environment.

# Highlights

## General

 * LXQt 1.0.0 depends on Qt 5.15, which is the last LTS version of Qt5.
 * Apart from bug fixes and workarounds, several functionalities are added to LXQt's file manager, like handling of emblems, new options in LXQt file dialog, an option to make desktop items sticky by default, recursive customization of folders, etc.
 * LXQt's image viewer has received several fixes and new options too.
 * The do-not-disturb mode is added to LXQt Desktop Notifications.
 * LXQt Panel has a new plugin, called "Custom Command", which does what its name says.
 * Saving and loading of Qt palettes are possible in LXQt Appearance Configuration.
 * Idleness checks can be paused from the tray icon of LXQt Power Manager.
 * Drag-and-drop file names are quoted in QTerminal.
 * Two LXQt themes are added and problems in the existing themes are fixed.
 * As always, translations have received many updates.
  * Other changes that can be found in change logs of LXQt components.

And so on...

**Reply by stefonarch** [reply]
I'd mention the smooth scroll too.

**Reply by tsujan** [reply]
> I'd mention the smooth scroll too.

I put it in "And so on..." but OK; will add it here too.

**Reply by tsujan** [reply]
The release is finished. If you find a problem, please tell me to fix it as soon as possible — hopefully, there isn't any.

As usual, I put the announcement in https://github.com/lxqt/lxqt/releases/tag/1.0.0. The user site (https://lxqt-project.org/) needs to be updated; otherwise, some users will try to kill us. @stefonarch, could you do it?

**Reply by stefonarch** [reply]
I never did postings, I'm not familiar with it. I could do it but have little time atm.
I see @luis-pereira did it in spring.

**Reply by tsujan** [reply]
Sorry, I thought you were familiar with it.

@luis-pereira, do you have time for it?

**Reply by stefonarch** [reply]
I could announce it shortly on twitter, or should I wait for the website?

**Reply by tsujan** [reply]
As you see fit. It *is* released. If you announce it before the website is updated, you could refer to https://github.com/lxqt/lxqt/releases/tag/1.0.0.

**Reply by stefonarch** [reply]
Done: https://twitter.com/lxqt_project

**Reply by yan12125** [reply]
As a record, I updated most LXQt-related packages on Arch. However, qtermwidget & qterminal are not updated yet due to incompatibility with the latest PyQt5, as mentioned at https://github.com/lxqt/qtermwidget/pull/449. I will try to find a solution in the near future.

**Reply by stefonarch** [reply]
Twitter probably is quite effective to spread news:
https://linuxstoney.com/lxqt-1-0-graphics-environment-released/
https://linuxnews.de/2021/11/lxqt-desktop-umgebung-erreicht-stabile-version-1-0/

**Reply by tsujan** [reply]
> However, qtermwidget & qterminal are not updated yet due to incompatibility with the latest PyQt5...

@yan12125, tell me if you need a point release for it.

**Reply by yan12125** [reply]
Thanks! Indeed having a point release will be better. I will tell you when fixes are ready.

**Reply by tsujan** [reply]
> Twitter probably is quite effective to spread news

Yes, I also saw 2 other articles in English.

BTW, the main screenshot at https://lxqt.github.io/ is beautiful. Please keep it when the site is updated (except for 0.17.0 in it).

**Reply by edwardp3** [reply]
PCLinuxOS updated their LXQt packages to 1.0.0 overnight, I updated them this morning and they work great. No issues.  Thank you to the Developers and Contributors for a job well done.

**Reply by MichaelTunnell** [reply]
I am glad to see the 1.0 milestone being reached, will promote in today's [TWIL](https://destinationlinux.network/shows/this-week-in-linux/)

**Reply by tsujan** [reply]
@edwardp3 
Thanks for the feedback! Although we always test the latest sources, we need to make sure that there is no issue.

**Reply by stefonarch** [reply]
> BTW, the main screenshot at https://lxqt.github.io/ is beautiful. Please keep it when the site is updated (except for 0.17.0 in it).

This was  @iamthesenate1  's work, based on previous screens, hopefully doing it again :)

**Reply by yan12125** [reply]
> qtermwidget & qterminal are not updated yet due to incompatibility with the latest PyQt5

Here comes [a hack](https://github.com/archlinux/svntogit-packages/commit/471adaa4b2d941f1c436fc49980f286253e7ba36) for pyqt5, and qtermwidget/qterminal are updated on Arch!

**Reply by maverick74** [reply]
> BTW, the main screenshot at https://lxqt.github.io/ is beautiful. Please keep it when the site is updated (except for 0.17.0 in it).

I just think that those pictures should show the main menu also open. 
It would transmit simplicity and functionality and make users feel right at home

**Reply by stefonarch** [reply]
I'm updating the screenshots and add some more and different ones too. 
I think we should prepare those and all the release post _before_ we release next time ;)

**Reply by tsujan** [reply]
> I think we should prepare those and all the release post before we release next time ;)

IMHO, it's very good to make releases on time, but it's also understandable that the site might be updated a few days later. Updating the site takes time (you know that better than I) and there's no guarantee that everyone has free time during the release.

Also, it's understandable that some users may forget that LXQt is made by volunteers, who have personal lives. It's obvious to us but, perhaps, not to all.

**Reply by maverick74** [reply]
@tsujan What?!?!?! 
You're not robots working on a mars cave 24/7 with the sole purpose to please our deepest wishes?!?!?
My heart is broken!!!
you lie for sure!
Next you'll say Santa isn't real and that unicorns don't live in our nearest forest!!!

**Reply by maverick74** [reply]
Naturally everyone loves updates, but even paid personal sometimes stoles too many time from their family for the "virtual world".

That is not fair for those around us who don't live "on a PC".

I think everyone understands that @tsujan ... and those who don't... well they'll have to live with it as everyone else ;)

**Reply by tsujan** [reply]
@maverick74
You won't believe me unless you handle reports. Of course, most people know that LXQt members are volunteers who dedicate parts of their time and energy to LXQt for free. Still some don't. And LXQt isn't alone in this; I've seen similar comments in KDE bug tracker.

**Reply by stefonarch** [reply]
> IMHO, it's very good to make releases on time, but it's also understandable that the site might be updated a few days later. Updating the site takes time (you know that better than I) and there's no guarantee that everyone has free time during the release.

Sure, but I think at least for a main posting on the website we'll find the 5 minutes necessary. I happened to notice  some discussions where people said "I cannot find any release news on their website".

**Reply by maverick74** [reply]
> You won't believe me unless you handle reports... Still some don't.

Weird time we live on...........

**Reply by tsujan** [reply]
> I happened to notice some discussions where people said "I cannot find any release news on their website".

IMO, you updated it fast.

**Reply by stefonarch** [reply]
In spring I will do it even faster ;)

**Comment by stefonarch** [comment]
[This](https://github.com/lxqt/lxqt/wiki/KF5-usage-in-LXQt) wiki page needs probably some fixes:
> Targeting for 1.0: Move dependency out of liblxqt and into apps that hard require it

**Reply by tsujan** [reply]
> This wiki page needs probably some fixes:

Yes, it does. It especially will when KF6 comes out and we use it.
